File: app/src/main/java/com/hnidesu/taskmanager/utility/HashUtil.java
package com.hnidesu.taskmanager.utility;

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.util.zip.CRC32;

public class HashUtil {
    public static String MD5Digest(String text) throws Exception {
        MessageDigest md5= MessageDigest.getInstance("MD5");
        md5.update(text.getBytes(StandardCharsets.UTF_8));

        StringBuilder sb=new StringBuilder();
        for (byte b:md5.digest()){
            String hex=String.format("%X",b);
            if(hex.length()==1)
                hex="0"+hex;
            sb.append(hex);
        }
        return sb.toString();
    }
    public static long CRC32Digest(CharSequence text) {
        CRC32 crc32=new CRC32();
        crc32.update(text.toString().getBytes(StandardCharsets.UTF_8));
        return crc32.getValue();
    }
}

File: app/src/main/java/com/hnidesu/taskmanager/util/HashUtil.kt
package com.hnidesu.taskmanager.util

import java.nio.charset.StandardCharsets
import java.security.MessageDigest
import java.util.zip.CRC32

object HashUtil{

    fun md5Digest(text: String): String {
        val md5 = MessageDigest.getInstance("MD5")
        md5.update(text.toByteArray(StandardCharsets.UTF_8))
        val sb = StringBuilder()
        for (b in md5.digest()) {
            var hex = String.format("%X", arrayOf(b))
            if (hex.length == 1) {
                hex = "0$hex"
            }
            sb.append(hex)
        }
        return sb.toString()
    }

    fun crc32Digest(text: String): Long {
        return CRC32().let {
            it.update(text.toByteArray(StandardCharsets.UTF_8))
            it.value
        }

    }

}
